batch/batch/front_end/front_end.py

```diff
@@ -338,52 +338,67 @@ async def _get_job_log_from_record(app, batch_id, job_id, record):
                 return None
             raise
 
-    if state in ('Error', 'Failed', 'Success'):
-        file_store: FileStore = app['file_store']
-        batch_format_version = BatchFormatVersion(record['format_version'])
+    if state in ('Pending', 'Ready', 'Creating'):
+        return None
 
-        async def _read_log_from_gcs(task):
-            try:
-                data = await file_store.read_log_file(
-                    batch_format_version, batch_id, job_id, record['attempt_id'], task
-                )
-            except FileNotFoundError:
-                id = (batch_id, job_id)
-                log.exception(f'missing log file for {id} and task {task}')
-                data = 'ERROR: could not find log file'
-            return task, data
+    if state == 'Cancelled' and record['last_cancelled_attempt_id'] is None:
+        return None
 
-        spec = json.loads(record['spec'])
-        tasks = []
+    assert state in ('Error', 'Failed', 'Success', 'Cancelled')
 
-        has_input_files = batch_format_version.get_spec_has_input_files(spec)
-        if has_input_files:
-            tasks.append('input')
+    batch_format_version = BatchFormatVersion(record['format_version'])
+    attempt_id = record['attempt_id'] or record['last_cancelled_attempt_id']
+    assert attempt_id
```

Comment:
prefer is not None because if we ever make attempt ids that are false-y this will fail.